src/iters.h

```diff
@@ -22,7 +23,7 @@ template <typename F, typename Iter>
 struct TransformIterator {
  private:
   Iter iter;
-  F f;
+  std::optional<F> f;
```

Comment:
This is optional so we can use the `emplace` function to avoid calling the copy assignment operator of `f`, which doesn't exist for lambdas. They only have copy constructors.